zyte_spider_templates/spiders/ecommerce.py

```diff
@@ -290,16 +330,30 @@ def parse_search_request_template(
         if probability is not None and probability <= 0:
             return
         for query in self.args.search_queries:
+            meta: Dict[str, Any] = {
+                "crawling_logs": {"page_type": "productNavigation"},
+            }
+            if self.args.dont_follow_product_links:
+                meta["inject"] = [ProductList]
+                if self._custom_attrs_dep:
+                    meta["inject"].append(self._custom_attrs_dep)
             yield search_request_template.request(query=query).to_scrapy(
                 callback=self.parse_navigation,
-                meta={
-                    "crawling_logs": {"page_type": "productNavigation"},
-                },
+                meta=meta,
             )
 
     def parse_navigation(
```

Comment:
It seems you're still following product links in this callback, even when self.args.dont_follow_product_links is True.